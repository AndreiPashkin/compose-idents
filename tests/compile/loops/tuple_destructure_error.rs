//! Tests incorrect tuple destructuring producing correctly formatted compile-time errors.
use compose_idents::compose_idents;

compose_idents!(
    for (a, (b, c)) in [(foo, (bar, baz)), (gork, bork)]

    my_fn = concat(a, b, c),
    {
        fn my_fn() -> u32 { 42 }
    }
);

fn main() {}
