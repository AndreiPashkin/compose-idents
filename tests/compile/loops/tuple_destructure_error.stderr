error: Mismatched number of elements in the tuple
 --> tests/compile/loops/tuple_destructure_error.rs:5:44
  |
5 |     for (a, (b, c)) in [(foo, (bar, baz)), (gork, bork)]
  |                                            ^
